app/models/term.rb

```diff
@@ -11,7 +11,13 @@
 #  updated_at :datetime         not null
 #
 class Term < ApplicationRecord
-  
-  scope :current, -> { where(':date BETWEEN term_start AND term_end', date: Date.today)}
+
+  validates_presence_of :code, :name, :term_start, :term_end
+
+  scope :sorted, -> { where.not(term_start: nil).order(:term_start, term_end: :desc) }
+  scope :between_start_and_end_dates, -> { sorted.where(':date BETWEEN term_start AND term_end', date: Date.today) }
+  scope :started_before_today, -> { sorted.where('term_start < :date', date: Date.today) }
+  scope :max_started_before_today, -> { sorted.where(term_start: started_before_today.max.term_start) }
```

Comment:
I would like to make a suggestion here. Please correct me if I am wrong. If the today's date is outside a term, then the current logic is to display programs of the recently completed term. I think in such a scenario, we should not display any programs and render text like "No programs for current date / term". The user can then select required term from the dropdown to view relevant programs.

Reply:
We should ask Sarah and Sheri about that. They are going to use the application.

Reply:
ok thanks.

Reply:
@anantaUM I returned the Term current scope to where(':date BETWEEN classes_begin_date AND classes_end_date', date: Date.today). 
I edited the Programs index page to display a message if the current date is between terms or if there are no programs for the selected term.
I talked to Sarah and Sheri about start and end terms dates. Sarah doesn't care and Sheri wants it to be classes begin and end.
They are both concerned about being able to access programs between terms. That is not a problem, they can always access all programs when they select a term.
Please review the PR again.